// ngx-mat-timepicker/esm2020/lib/services/ngx-mat-timepicker-locale.service.mjs
import { Inject, Injectable } from "@angular/core";
//
import { NGX_MAT_TIMEPICKER_LOCALE } from "../tokens/ngx-mat-timepicker-time-locale.token";
import * as i0 from "@angular/core";
export class NgxMatTimepickerLocaleService {
    constructor(initialLocale) {
        this._locale = initialLocale;
    }
    get locale() {
        return this._locale;
    }
    updateLocale(newValue) {
        this._locale = newValue || this._initialLocale;
    }
}
NgxMatTimepickerLocaleService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxMatTimepickerLocaleService, deps: [{ token: NGX_MAT_TIMEPICKER_LOCALE }], target: i0.ɵɵFactoryTarget.Injectable });
NgxMatTimepickerLocaleService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxMatTimepickerLocaleService, providedIn: "root" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxMatTimepickerLocaleService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: "root"
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [NGX_MAT_TIMEPICKER_LOCALE]
                }] }]; } });
